Facet Search API (#246)
meilisearch/specifications: text/0246-facet-search-api.md

# Facet Search API

## 1. Summary

The facet search endpoint retrieves facet values from a query.

## 2. Motivation

> When many values exist for a facet, end-users need to be able to discover non-show values they can select in order to refine their faceted search. Being able to search for values for a facet solves this need.

Reviewer: ```suggestion
When a facet has lots of values, end-users need to be able to find specific values that are not returned if there are too many or lost in a sea of other values. Being able to search for specific values of a facet solves this need.
```

### 3.1. Facet Search Payload Parameters

| Field                                                 | Type                     | Required |
|-------------------------------------------------------|--------------------------|----------|
| [`facetName`](#311-facetName)                         | String                   | True     |
| [`facetQuery`](#312-facetQuery)                       | String                   | False    |

#### 3.1.1. `facetName`

- Type: String
- Required: True
- Default: `null`

`facetName` contains the facet name to search values on.

- 🔴 Omitting `facetName` returns a `missing_facet_search_facet_name`(0061-error-format-and-definitions.md#missing_facet_search_facet_name) error.
- 🔴 Sending a value with a different type than `String` for `facetName` returns a [missing_facet_search_facet_name](0061-error-format-and-definitions.md#minvalid_facet_search_facet_name) error.
- 🔴 Sending a field not defined as a `filterableAttributes` for `facetname` returns an [invalid_facet_search_facet_name](0061-error-format-and-definitions.md#invalid_facet_search_facet_name) error.

#### 3.1.2. `facetQuery`

- Type: String
- Required: False
- Default: `null`

> `facetQuery` contains the terms to search within the facet values contained in the index documents.

Reviewer: ```suggestion
`facetQuery` contains the terms to search within the facet values of the `facetName`.
```
